### PR TITLE
[FIX] hr_timesheet

### PR DESCRIPTION
**Pull request by nboulif**
ENT : https://github.com/odoo/enterprise/pull/30271

### CODE REVIEW

**Comment by nboulif** [comment]
@fw-bot up to 15.0

**Comment by nboulif** [comment]
@xavierbol Indeed, I missed this one, thanks! I updated the PRs :+1:

**Comment by nboulif** [comment]
@fw-bot up to 14.0

File: addons/hr_timesheet/models/hr_timesheet.py [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
+from collections import defaultdict
```
Comment (IT-Ideas) [inline addons/hr_timesheet/models/hr_timesheet.py:4]: Maybe remove the last 4 lines of the commit message and replace them with:
```
backport of odoo/odoo#81693
```

File: addons/hr_timesheet/models/hr_timesheet.py [hunk 2]
```diff
@@ -46,7 +46,7 @@ def _domain_task_id(self):
         'project.project', 'Project', compute='_compute_project_id', store=True, readonly=False,
         domain=_domain_project_id)
     user_id = fields.Many2one(compute='_compute_user_id', store=True, readonly=False)
-    employee_id = fields.Many2one('hr.employee', "Employee", domain=_domain_employee_id)
+    employee_id = fields.Many2one('hr.employee', "Employee", domain=_domain_employee_id, context={'active_test': False})
```
Comment (IT-Ideas) [inline addons/hr_timesheet/models/hr_timesheet.py:49]: Same idea here, replace the three last lines of the commit by:
```
backport of odoo/odoo#85481
```

File: addons/hr_timesheet/models/hr_timesheet.py [hunk 3]
```diff
@@ -102,16 +102,19 @@ def create(self, vals_list):
 
         # Although this make a second loop on the vals, we need to wait the preprocess as it could change the company_id in the vals
         # TODO To be refactored in master
-        company_ids_in_vals = list({vals['company_id'] for vals in vals_list if vals.get('company_id', False)})
-        employees = self.env['hr.employee'].with_context(active_test=False).search([('user_id', 'in', user_ids), ('company_id', 'in', [self.env.company.id] + company_ids_in_vals)])
-        user_map = defaultdict(dict)
+        employees = self.env['hr.employee'].sudo().with_context(active_test=False).search([('user_id', 'in', user_ids)])
```
Comment (IT-Ideas) [inline addons/hr_timesheet/models/hr_timesheet.py]: Same idea here to, could you replace the 4 last lines by:
```
backport of odoo/odoo#85780
```

File: addons/hr_timesheet/i18n/hr_timesheet.pot [hunk 4]
```diff
@@ -971,6 +971,12 @@ msgstr ""
 msgid "Timesheets for sub-task:"
 msgstr ""
 
+#. module: hr_timesheet
```
Comment (IT-Ideas) [inline addons/hr_timesheet/i18n/hr_timesheet.pot:974]: Same here too, could you replace the 4 last lines by:
```
backport of odoo/odoo#94308
```

**Review by IT-Ideas — APPROVED** [review]
Thanks for the backporting of the fix 🙂 
Just a remark on the commit messages to clean a bit in order to make it clearer that they are backports.

**Comment by nboulif** [comment]
@IT-Ideas My bad :sweat_smile: Thanks for review! :pray:

**Comment by nle-odoo** [comment]
robodoo delegate+

**Comment by nboulif** [comment]
robodoo r+

**Comment by nboulif** [comment]
robodoo rebase-ff

**Pull request closed without merging**